e2e/src/common/utils.ts

```diff
@@ -191,6 +197,54 @@ export async function waitForEvents<
   return events;
 }
 
+// Currently only handle simple single return types - uint256 | bytesX | string | bool
+export async function pollForContractMethodReturnValue<
+  ExpectedReturnType extends bigint | string | boolean,
+  R extends [ExpectedReturnType],
+>(
+  method: TypedContractMethod<[], R, "view">,
+  expectedReturnValue: ExpectedReturnType,
+  pollingInterval: number = 500,
+  timeout: number = 2 * 60 * 1000,
+): Promise<boolean> {
+  let isExceedTimeOut = false;
+  setTimeout(() => {
+    isExceedTimeOut = true;
+  }, timeout);
+
+  while (!isExceedTimeOut) {
+    const returnValue = await method();
+    if (returnValue === expectedReturnValue) return true;
+    await wait(pollingInterval);
+  }
+
+  return false;
+}
+
+// Currently only handle single uint256 return type
+export async function pollForContractMethodReturnValueExceedTarget<
+  ExpectedReturnType extends bigint,
+  R extends [ExpectedReturnType],
+>(
+  method: TypedContractMethod<[], R, "view">,
+  targetReturnValue: ExpectedReturnType,
+  pollingInterval: number = 500,
+  timeout: number = 2 * 60 * 1000,
+): Promise<boolean> {
+  let isExceedTimeOut = false;
+  setTimeout(() => {
+    isExceedTimeOut = true;
+  }, timeout);
+
+  while (!isExceedTimeOut) {
+    const returnValue = await method();
+    if (returnValue >= targetReturnValue) return true;
+    await wait(pollingInterval);
+  }
+
+  return false;
+}
+
```

Comment:
Can we generalize the above two functions into the sth like the following:
```
// Currently only handle simple single return types - uint256 | bytesX | string | bool
export async function pollForContractMethodReturnValue<
  ExpectedReturnType extends bigint | string | boolean,
  R extends [ExpectedReturnType],
>(
  method: TypedContractMethod<[], R, "view">,
  expectedReturnValue: ExpectedReturnType,
  compareFunc: (a: ExpectedReturnType, b: ExpectedReturnType) => Boolean = (a,b) => {return a === b},
  pollingInterval: number = 500,
  timeout: number = 2 * 60 * 1000,
): Promise<boolean> {
  let isExceedTimeOut = false;
  setTimeout(() => {
    isExceedTimeOut = true;
  }, timeout);

  while (!isExceedTimeOut) {
    const returnValue = await method();
    if (compareFunc(returnValue, expectedReturnValue)) return true;
    await wait(pollingInterval);
  }

  return false;
}
```
such that we could inject the comparison function (default to equal) and hence can be used like the following:
```
const isBlockFinalized = await pollForContractMethodReturnValue(
      lineaRollupV6.currentL2BlockNumber,
      BigInt(blockNumberToCheckFinalization),
      (returnVal, expectedVal) => { return returnVal >= expectedVal}
    );
```

Reply:
good point, updated